src/main/resources/public/index.js

```diff
@@ -0,0 +1,59 @@
+const StoreList = {
+    data() {
+        return {
+            stores: [],
+            currentStore: null,
+            currentPage: 1,
+            nameSearch: null,
+            newStoreName: null,
+        }
+    },
+    watch: {
+        nameSearch() {
```

Comment:
is there any improvement that you could do here?

Reply:
I could have spared a lot of network calls by applying debouncing or throttling to the search.